airbyte-integrations/connectors/source-mailchimp/source_mailchimp/source.py

```diff
@@ -21,67 +21,201 @@
+        if next_page_token:
+            params.update(next_page_token)
+        return params
+
+    def request_headers(self, **kwargs) -> Mapping[str, Any]:
+        return {}
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        response_json = response.json()
+        yield from response_json[self.data_field]
+
+
+class IncrementalMailChimpStream(MailChimpStream, ABC):
+    state_checkpoint_interval = math.inf
+
+    @property
+    @abstractmethod
+    def cursor_field(self) -> str:
+        """
+        Defining a cursor field indicates that a stream is incremental, so any incremental stream must extend this class
+        and define a cursor field.
+        """
+        pass
+
+    def get_updated_state(self, current_stream_state: MutableMapping[str, Any], latest_record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        Return the latest state by comparing the cursor value in the latest record with the stream's most recent state object
+        and returning an updated state object.
+        """
+
+        zero_date = datetime(1900, 1, 1)
+
+        current_state = current_stream_state.get(self.cursor_field)
+        latest_state = latest_record.get(self.cursor_field)
+        if not any([current_state, latest_state]):  # noqa
+            return {self.cursor_field: zero_date}
+
+        if not all([current_state, latest_state]):
+            nonzero_state = list(filter(bool, [current_state, latest_state]))[0]
+            if not isinstance(nonzero_state, datetime):
+                nonzero_state = dateutil.parser.isoparse(nonzero_state)
+            return {self.cursor_field: nonzero_state}
+        else:
+            if not isinstance(current_state, datetime):
+                current_state = dateutil.parser.isoparse(current_state)
+            if not isinstance(latest_state, datetime):
+                latest_state = dateutil.parser.isoparse(latest_state)
+            max_value = max(latest_state, current_state)
+            return {self.cursor_field: max_value}
+
+    def request_params(self, stream_state=None, **kwargs):
+
+        stream_state = stream_state or {}
+        params = super().request_params(stream_state=stream_state, **kwargs)
+        default_params = {"sort_field": self.cursor_field, "sort_dir": "ASC"}
+        since_value = stream_state.get(self.cursor_field)
+        if isinstance(since_value, datetime):
+            since_value = since_value.isoformat()
+        default_params[f"since_{self.cursor_field}"] = since_value
+        params.update(default_params)
+        return params
+
+
+class Lists(IncrementalMailChimpStream):
+    cursor_field = "date_created"
+    name = "Lists"
+    data_field = "lists"
+
+    def path(self, **kwargs) -> str:
+        return "lists"
+
+
+class Campaigns(IncrementalMailChimpStream):
+    cursor_field = "create_time"
+    name = "Campaigns"
```

Comment:
you don't need to override name here, it is deducted from the class name

Reply:
if you can't find shema files because they names capitalized, then it is better to rename schema files